pages/_app.js

```diff
@@ -32,13 +40,54 @@ async function fetchThisWeekAndReturnArray() {
   return formatedArray;
 }
 
+function addCountryFlagsToArray(events) {
```

Comment:
I think `addCountryFlagsToArray` is a misnomer, because you're not adding to an array, you're mutating every object in an array. You've already got everything you need (more or less) to avoid direct mutation and instead work with something called a dictionary or map. It roughly works like this:

```javascript
//This is our dictionary.

const countryFlags = {
USD: "🇺🇸",
EUR: "🇪🇺",
AUD: "🇦🇺",
}

//This is the equivalent of your `events` array.
const countries = [{name: "USA", currencyCode: "USD"}, 
{name: "Germany", currencyCode: "EUR"}, 
{name: "UK", currencyCode: "GBP"}]

//We're mapping over our countries and adding a flag for every country,
// if we're not finding the currency, we're returning null
const countriesWithFlags = countries.map((country) => 
({...country, countryFlag: countryFlags[country.currencyCode] || null}))

//Logging countriesWithFlags would return this:
[{name: "USA", currencyCode: "USD", countryFlag: "🇺🇸"}, 
{name: "Germany", currencyCode: "EUR", countryFlag: "🇪🇺"},
{name: "UK", currencyCode: "GBP", countryFlag: null}]
```

Reply:
makes sense, will implement it that way